# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xdm/startwm.patch
# Copyright (C) 2004 - 2023 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

More flexible root application startup.

--- xdm-1.1.9/config/Xsetup_0.vanilla   2009-09-19 12:44:22.000000000 +0200
+++ xdm-1.1.9/config/Xsetup_0   2009-09-19 12:44:26.000000000 +0200
@@ -1,2 +1,3 @@
 #!/bin/sh
 xconsole -geometry 480x130-0-0 -daemon -notify -verbose -fn fixed -exitOnFail
+xsetroot -solid black
--- xdm-1.1.12/config/Xsession.in.orig	2022-03-31 11:44:43.648642659 +0200
+++ xdm-1.1.12/config/Xsession.in	2022-03-31 11:47:20.338650742 +0200
@@ -62,5 +62,12 @@
 	if [ -r "$resources" ]; then
 		BINDIR/xrdb -load "$resources"
 	fi
+
+	[ -f /etc/profile ] && . /etc/profile
+
+	[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+	[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+	[ "$WINDOWMANAGER" != "twm"   ] && exec $WINDOWMANAGER
+	exec twm; exec xterm
 	exec BINDIR/xsm
 fi
